File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        left =0
        right = left+1
        maxprofit =0
        while right<len(prices):
            print(left, right ,prices[left] , prices[right])
            if prices[left]<prices[right]:
                maxprofit = max(maxprofit, prices[right]-prices[left])
                right+=1
            else:
                left=right
                right+=1

        return maxprofit

File: Data Structures & Algorithms/binary-tree-diameter/submission-1.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number}
     */
    diameterOfBinaryTree(root) {
        let diameter = 0;
        function findDiameter(root){
            if(root == null){
                return null;
            }
            let leftdepth = findDiameter(root.left);
            let rightdepth = findDiameter(root.right);
            diameter = Math.max(leftdepth + rightdepth ,diameter);
            return 1+Math.max(leftdepth,rightdepth);
        }
        findDiameter(root);
        return diameter;

    }
}
